library/src/webapp/js/lessons-subnav.js

```diff
@@ -83,399 +85,64 @@
+        url += '&itemId='+subpage.itemId;
         url += '&path=clear_and_push';
-        url += '&title=' + sub_page.name;
+        url += '&title=' + subpage.name;
         url += '&newTopLevel=false';
         return url;
     };
 
-
-    LessonsSubPageNavigation.prototype.set_current_for_page_id = function(page_id, context_id) {
-        var self = this;
-
-        if (self.data.hasOwnProperty(context_id)) {
-            self.data[context_id].forEach(function(sub_page) {
-                if (sub_page.sendingPage == page_id) {
-                    var li = sub_page.submenu_item;
-                    var parent = li.parentElement.parentElement;
-                    parent.classList.add('is-parent-of-current');
-                    li.classList.add('is-current');
-                }
-            });
-        }
+    LessonsSubPageNavigation.prototype.getCurrentPlacement = function() {
+        const url = new URL(window.location.href);
+        const parts = url.pathname.split('/');
+        return (parts.length >= 6) ? parts[5] : '';
     };
 
-    LessonsSubPageNavigation.prototype.set_current_for_item_and_page_id = function(item_id, page_id, context_id) {
-        var self = this;
-
-        if (self.data.hasOwnProperty(context_id)) {
-            self.data[context_id].forEach(function(sub_page) {
-                if (sub_page.sendingPage == page_id) {
-                    if (sub_page.itemId == item_id) {
-                        var li = sub_page.submenu_item;
-                        var parent = li.parentElement.parentElement;
-                        parent.classList.add('is-parent-of-current');
-                        li.classList.add('is-current');
-                    }
-                }
-            });
-        }
-    };
-
-
-    LessonsSubPageNavigation.prototype.set_current_for_item_id = function(item_id, context_id) {
-        var self = this;
-
-        if (self.data.hasOwnProperty(context_id)) {
-            self.data[context_id].forEach(function(sub_page) {
-                if (sub_page.itemId == item_id) {
-                    var li = sub_page.submenu_item;
-                    var parent = li.parentElement.parentElement;
-                    parent.classList.add('is-parent-of-current');
-                    li.classList.add('is-current');
-                }
-            });
+    LessonsSubPageNavigation.prototype.getSubpageElement = function() {
+        const lessonsSubnavToolIdInput = document.getElementById('lessonsSubnavToolId');
+        const lessonsSubnavPageIdInput = document.getElementById('lessonsSubnavPageId');
+        const lessonsSubnavItemIdInput = document.getElementById('lessonsSubnavItemId');
+        let subpageElement = null;
+
+        if ((lessonsSubnavToolIdInput !== null) && (lessonsSubnavPageIdInput !== null) &&
+            (lessonsSubnavItemIdInput !== null)) {
+            subpageElement = document.querySelector('#toolMenu a[href*="/tool/' + lessonsSubnavToolIdInput.value + 
```

Comment:
This worries me. This selector just seems so coupled to Lessons, for something that is part of the portal nav. If somebody decided to mess with those lessons specific urls in future, would they remember to come back to the portal code and update this selector?

Reply:
_Understood, though the dependency at root here is not new, as I think you’re aware. It’s just adapted from morpheus to the trinity context._ 

The three hidden fields that the highlighted condition is testing for is to assess whether or not we’re on a Lessons page rendered by the ShowPage template. As-is, that’s the only template from which we can glean what the current Lessons page is for us (in the portal) to highlight in the sitenav. In other words there is no continuity for displaying those hidden fields in the DOM across the Lessons tool for the other 20+ Lessons templates (e.g., ShowItem, Reorder, IndexOfPages, etc.), so the client side portal code resorts to highlighting the “Main Page” based on the 

_> If somebody decided to mess with those lessons specific urls in future, would they remember to come back to the portal code and update this selector?_

If somebody did change the URLs, our institution would certainly notice-- per the conversation in the jira. :) But yes, point taken.